ipv-core/src/main/java/com/ibm/whc/deid/providers/masking/BinningMaskingProvider.java

```diff
@@ -55,13 +68,27 @@ public String mask(String identifier) {
       debugFaultyInput("identifier");
       return null;
     }
-
+    
     double value;
     try {
       value = Double.parseDouble(identifier);
     } catch (NumberFormatException e) {
       // For this provider, we do not return a random value
       return applyUnexpectedValueHandling(identifier, null);
+    }  
+
+    if (this.useSingleBucketOverThreshold==true && this.useSingleBucketUnderThreshold==true) {
```

Comment:
We want to allow either of the thresholds or both or neither to be true.  This way the both have to be true for either to be used.  I suggest having one check for the over and another completely separate check for the under.  That way they can work independently.